File: config/python.py
""" python deps for this project """

import config.shared

install_requires: list[str] = [
    "biopython",
    "numpy",
    "pandas",
    "scipy",
    "matplotlib",
    "seaborn",
]
build_requires: list[str] = config.shared.BUILD
test_requires: list[str] = config.shared.TEST
types_requires: list[str] = [
    "types-PyYAML",
    "pandas-stubs",
]
requires = install_requires + build_requires + test_requires + types_requires

File: config/python.lua
-- python deps for this project

dofile("config/shared.lua")

-- append every element of "src" onto "dst"
local function extend(dst, src)
    for _, value in ipairs(src) do
        table.insert(dst, value)
    end
    return dst
end

INSTALL_REQUIRES = {
    "biopython",
    "numpy",
    "pandas",
    "scipy",
    "matplotlib",
    "seaborn",
}
BUILD_REQUIRES = BUILD
TEST_REQUIRES = TEST
TYPES_REQUIRES = {
    "types-PyYAML",
    "pandas-stubs",
}

REQUIRES = {}
extend(REQUIRES, INSTALL_REQUIRES)
extend(REQUIRES, BUILD_REQUIRES)
extend(REQUIRES, TEST_REQUIRES)
extend(REQUIRES, TYPES_REQUIRES)
